File: python/estimators/est_octahedron.py
import numpy as np

from .est_donutLSQ1_2D import est_donutLSQ1_2D
from .est_qLSQiter1D import est_qLSQiter1D

def est_octahedron(photonsi, patternposi, L, fwhm, background=0):
    photonsxy = photonsi[0:4]
    photonsz = photonsi[4:6]
    patternposxy = patternposi[0:4,:]

    if len(photonsi) == 7:
        # center probed
        photonsxy = np.vstack([photonsxy, photonsi[6]])
        photonsz = np.vstack([photonsz, photonsi[6]])
        patternposxy = np.vstack([patternposxy,
                                  np.zeros(patternposxy.shape[1], 
                                           dtype=patternposxy.dtype)])

    xest = est_donutLSQ1_2D(photonsxy, patternposxy, L, fwhm, background)
    xest[2] = est_qLSQiter1D(photonsz, L)

    return xest

File: MATLAB/estimators/est_octahedron.m
function xest=est_octahedron(photonsi,patternposi,L,fwhm,background)
if nargin<5
    background=0;
end
photonsxy=photonsi(1:4);
photonsz=photonsi(5:6);
patternposxy=patternposi(1:4,:);
if length(photonsi)==7 %center probed
    photonsxy(end+1)=photonsi(7);
    photonsz(end+1)=photonsi(7);
    patternposxy(end+1,:)=0;
end
xest=est_donutLSQ1_2D(photonsxy,patternposxy,L,fwhm,background);
xest(3)=est_qLSQiter1D(photonsz,L);

end